pkg/auth/spire/certificate_provider.go

```diff
@@ -67,6 +68,16 @@ func (s *SpireDelegateClient) sniToSPIFFEID(id identity.NumericIdentity) string
 	return "spiffe://" + s.cfg.SpiffeTrustDomain + "/cilium-id/" + id.String()
 }
 
+func (s *SpireDelegateClient) spiffeIDToNumericIdentity(spiffeID string) (identity.NumericIdentity, error) {
+	prefix := "spiffe://" + s.cfg.SpiffeTrustDomain + "/cilium-id/"
```

Comment:
Given the discussion we had about maybe changing this, should we make this a const somewhere? I think it's in more than just here now?

Reply:
yup, I have cleaned this up in #24765